File: C++/TD5/Personne/etudiant.cpp
#include "etudiant.h"

Etudiant::Etudiant(string nom, string adresse, float moyenne):Personne(nom, adresse) {
	m_moyenne = moyenne;
}

float Etudiant::getMoyenne() {
	return m_moyenne;
}

void Etudiant::setMoyenne(float nMoyenne) {
	m_moyenne = nMoyenne;
}

void Etudiant::afficher() {
	cout << "Nom : " << getNom() << endl << "Adresse : " << getAdresse() << endl << "Moyenne : " << m_moyenne << endl;
}

File: Java/TD2/src/Etudiant.java
class Etudiant extends Personne {
    float m_moyenne = 0;

    public Etudiant() {
        super();
    }

    public Etudiant(String m_nom, String m_adresse, float m_moyenne) {
        super(m_nom, m_adresse);
        this.m_moyenne = m_moyenne;
    }

    public float getMoyenne() {
        return m_moyenne;
    }

    public void setMoyenne(float m_moyenne) {
        this.m_moyenne = m_moyenne;
    }

    @Override
    public String afficher() {
        String affichage;
        affichage = "Élève : " + m_nom + "\n" + "Adresse : " + m_adresse + "\n";
        affichage += "Moyenne : " + m_moyenne;
        return affichage;
    }
}
